### ISSUE TITLE
Training a skill not present in SDE aborts character update

### ISSUE THREAD

**Issue by BuumEdzott** [skyride/evething-2#6]
If a character is training a skill not present in the SDE (for example, the moon mining skills at present), the character_info task raises an exception and prevents the updating of _any_ of that character's information.

```
[2017-10-06 15:58:26,305: ERROR/MainProcess] Task thing.esi.character_info[9bdf2e23-c55e-4e8a-8645-9c2026fac9c7] raised unexpected: IntegrityError(1452, 'Cannot add or update a child row: a foreign key constraint fails (`evething2`.`thing_characterskill`, CONSTRAINT `thing_character_skill_id_6095e43f33a8c14f_fk_thing_skill_item_id` FOREIGN KEY (`skill_id`) REFERENCES `thing_skill` (`item_id`))')
Traceback (most recent call last):
  File "/efs/evething2env/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/efs/evething2env/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/transaction.py", line 394, in inner
    return func(*args, **kwargs)
  File "thing/tasks/esi/characterinfo.py", line 139, in run
    db_skill.save()
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/base.py", line 617, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/base.py", line 698, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/base.py", line 731, in _do_insert
    using=using, raw=raw)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/query.py", line 921, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 921, in execute_sql
    cursor.execute(sql, params)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/efs/evething2env/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 129, in execute
    return self.cursor.execute(query, args)
  File "/efs/evething2env/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/efs/evething2env/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
```

**Comment by skyride** [comment]
I've looked at resolving this in the past, but honestly it's a good reminder to get things updated.

I think the solution is to introduce a regular check to see if the SDE is up to date. I actually just introduced the SDE to avrseauth. The way it works there is there's a command that downloads the sqlite distribution if the SDE from fuzzworks, then another command that imports from it. I could probably make it grab the MD5 hash file of the file for latest when it runs the import, then it can just check against the version every 24hr or something and display a reminder.
